Remaining tweaks to General Tools (#3411)
qgis/QGIS-Documentation: source/docs/user_manual/introduction/general_tools.rst

  Given that whenever you apply modifications to the layer properties, changes
  are stored in the active style, always ensure you are editing the right style
  to avoid mistakenly altering a style used in a :ref:`map theme <map_themes>`.

.. tip:: **Manage styles from layer context menu**

>    Right-click on the layer in the :guilabel:`Layers Panel` to add, rename

Reviewer: ```suggestion
   Right-click on the layer in the :guilabel:`Layers` panel to add, rename
```
Panel is no longer part of the GUI label

in another project.

Save as text file
........................

Clicking the |selectString| :menuselection:`Style --> Save Style`, you can
save the style as a:

* QGIS layer style file (:file:`.qml`)
* SLD file (:file:`.sld`), only available for vector layers

Used on file-based format layers (:file:`.shp`, :file:`.tab`...), :guilabel:`Save
as Default` generates a :file:`.qml` file for the layer (with the same name).
SLDs can be exported from any type of renderer -- single symbol,
categorized, graduated or rule-based -- but when importing an SLD, either a
single symbol or rule-based renderer is created.
This means that categorized or graduated styles are converted to rule-based.
If you want to preserve those renderers, you have to use the QML format.
On the other hand, it can be very handy sometimes to have this easy way of
converting styles to rule-based.

Save in database
.................

Vector layer styles can also be stored in a database if the layer datasource is a
database provider. Supported formats are PostGIS, GeoPackage, SpatiaLite, MSSQL
and Oracle. The layer style is saved inside a table (named :file:`layer_styles`)

> in the database. Click on :menuselection:`Save Style --> Save in database`

Reviewer: ```suggestion
in the database. Click on :menuselection:`Save Style... --> Save in database`
```
That's not the purpose of this PR but I'm afraid all this "Storing Styles in a File or a Database" section is outdated. eg there no more "Save in database" entry but you get a dialog in which you select either qml, sld, or db, with style's properties.

Author: Sure - a job for a separate PR, I guess.

then fill in the dialog to define a style name, add a description, a :file:`.ui`
file if applicable and to check if the style should be the default style.

You can save several styles for a single table in the database. However, each
table can have only one default style. Default styles can be saved in the layer

> database or in the QGIS local database, a SQLite database in the :file:`~/.qgis2/`

Reviewer: another qgis2 we should remove. Sorry I don't know what's the replacement.

Reviewer 2: Location depends on the operating system.

Reviewer: I think the styles db is now in the QGIS user profile and while the location of the profile folder depends on OS, the path to the db in the folder is the same for all and mentioning user profile (if confirmed) here should be a minimum (in any case, we must not keep .qgis**2**)
User profiles folders are/should be described elsewhere.

Author: ```suggestion
database.
```

Reviewer: I's not the layer database.
We have two different things here when you press "save as default" button from a database layer; you get a pop-up dialog with options to save style to "datasource database", "local database" or "cancel". And the differences between the first two options need to be clarified.

> directory (where QGIS stores its local settings).

Author: ```suggestion
```

.. note::

  You may encounter issues restoring the :file:`layer_styles` table from a
  PostgreSQL database backup. Follow :ref:`layer_style_backup` to fix that.

Load style
...........

When loading a layer in QGIS, if a default style already exists for this layer,
QGIS loads the layer with this style. Also :menuselection:`Style --> Restore Default`
looks for and loads that file, replacing the layer's current style.

:menuselection:`Style --> Load Style` helps you apply any saved style to a
layer. While text-file styles (:file:`.sld` or :file:`.qml`) can be applied
to any layer whatever its format, loading styles stored in a database is only
possible if the layer is from the same database or the style is stored in the
QGIS local database.

> The :guilabel:`Load Style from Database` dialog displays a list of styles related

Reviewer: ```suggestion
The :guilabel:`Database Styles Manager` dialog displays a list of styles related
```
While waiting for the section update let's use actual name

Variables can be defined at the application's global level, project level,
layer level, layout level, and layout item's level. Just like CSS
cascading rules, variables can be overwritten - e.g., a project level
variable will overwrite any application global level variables set with
the same name. You can use these variables to build text strings or other
custom expressions using the ``@`` character before the variable name. For
example in print layout creating a label with this content:

  This map was made using QGIS [% @qgis_version %]. The project file for this
  map is: [% @project_path %]

Will render the label like this::

>   This map was made using QGIS 3.4.4. The project file for this map is:

Reviewer: ```suggestion
  This map was made using QGIS 3.4.4-Madeira. The project file for this map is:
```
Looks like qgis_version takes name now
